### PR TITLE
fix: skip comments when loading test relationships

### PR DESCRIPTION
**Pull request by bryanhuhta**
Fixes https://github.com/authzed/spicedb/issues/329

From a brief test in the playground, comments in the Test Relationships are only of the format `// my comment` and not `/** my comment */`. This may need to be fact check though 😁

### CODE REVIEW

**Comment by josephschorr** [comment]
> From a brief test in the playground, comments in the Test Relationships are only of the format `// my comment` and not `/** my comment */`. This may need to be fact check though 😁

Yeah, that's correct :)

**Comment by bryanhuhta** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/validationfile/loader.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func PopulateFromFiles(ds datastore.Datastore, filePaths []string) (*FullyParsed
 			lines := strings.Split(relationships, "\n")
 			for index, line := range lines {
 				trimmed := strings.TrimSpace(line)
-				if len(trimmed) == 0 {
+				if len(trimmed) == 0 || strings.HasPrefix(trimmed, "//") {
```
Comment (josephschorr) [inline pkg/validationfile/loader.go:97]: Mind adding a test into `loader_test.go`?

**Comment by bryanhuhta** [comment]
When writing the tests, a few thoughts came to me:

- Is there any recommendation around using `testdata/` to store test data?
- I used `memdb.NewMemdbDatastore` to build a `datastore.Datastore` for my test. I'm not sure if there's an existing `datastore.Datastore` implementation specifically for being mocked in tests.

**Comment by josephschorr** [comment]
> When writing the tests, a few thoughts came to me:
> 
> * Is there any recommendation around using `testdata/` to store test data?

We actually do something similar for the consistency tests: https://github.com/authzed/spicedb/blob/main/internal/services/consistency_test.go (it loads from a directory called `testconfigs`), so follow the style in there. If the test data is small enough, you can also inline.

> * I used `memdb.NewMemdbDatastore` to build a `datastore.Datastore` for my test. I'm not sure if there's an existing `datastore.Datastore` implementation specifically for being mocked in tests.

MemDB is the correct datastore to use. See https://github.com/authzed/spicedb/blob/main/internal/services/consistency_test.go#L64 for an example

File: pkg/validationfile/loader_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	tests := []struct {
+		name      string
+		filePaths []string
+		want      []*v0.RelationTuple
+	}{
+		{
+			name:      "no comment",
+			filePaths: []string{"testdata/loader_no_comment.yaml"},
+			want: []*v0.RelationTuple{
+				{
```
Comment (josephschorr) [inline pkg/validationfile/loader_test.go:21]: You can use `tuple.Parse` to make this more compact (not required, but makes it a bit more readable)

File: pkg/validationfile/testdata/loader_with_comment.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      permission write = writer + admin
+      permission admin = owner
+  }
+relationships: >-
+  // Some comment
+
+  example/project:pied_piper#owner@example/user:milburga
+
+  example/project:pied_piper#reader@example/user:tarben
+
```
Comment (josephschorr) [inline pkg/validationfile/testdata/loader_with_comment.yaml:20]: Please add another comment here as well

**Review by josephschorr — DISMISSED** [review]
LGTM

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**